file/utils/init_provider_test.go

```diff
@@ -0,0 +1,261 @@
+	"os"
+	"path/filepath"
+	"testing"
+
+	vpc_prov "github.com/IBM/ibmcloud-volume-file-vpc/file/provider"
+	vpcconfig "github.com/IBM/ibmcloud-volume-file-vpc/file/vpcconfig"
+	"github.com/IBM/ibmcloud-volume-interface/config"
+	"github.com/IBM/ibmcloud-volume-interface/lib/provider"
+	auth "github.com/IBM/ibmcloud-volume-interface/provider/auth"
+	"github.com/IBM/ibmcloud-volume-interface/provider/local/fakes"
+	"github.com/IBM/secret-utils-lib/pkg/k8s_utils"
+	"github.com/stretchr/testify/assert"
+	"go.uber.org/zap"
+)
+
+const (
+	TestProviderAccessToken = "test-provider-access-token"
+	TestIKSAccountID        = "test-iks-account"
+)
+
+func TestInitProviders(t *testing.T) {
+	logger := zap.NewNop()
+	vpcfileconf := &vpcconfig.VPCFileConfig{}
+	k8sClient, _ := k8s_utils.FakeGetk8sClientSet()
+	pwd, _ := os.Getwd()
+
+	clusterConfPath := filepath.Join(pwd, "..", "..", "test-fixtures", "valid", "cluster_info", "cluster-config.json")
+	_ = k8s_utils.FakeCreateCM(k8sClient, clusterConfPath)
+
+	secretConfPath := filepath.Join(pwd, "..", "..", "test-fixtures", "slconfig.toml")
+	_ = k8s_utils.FakeCreateSecret(k8sClient, "DEFAULT", secretConfPath)
+
+	// Define test cases
+	testCases := []struct {
+		name           string
+		expectedErrMsg string
+		vpcConfig      *config.VPCProviderConfig
+		iksConfig      *config.IKSConfig
+		client         *k8s_utils.KubernetesClient
+	}{
+		{
+			name:           "Both VPC and IKS providers are enabled",
+			client:         &k8sClient,
+			expectedErrMsg: "",
+			// Setup test environment
+			vpcConfig: &config.VPCProviderConfig{
+				Enabled:       true,
+				VPCVolumeType: "test-vpc-volume-type",
+			},
+			iksConfig: &config.IKSConfig{
+				Enabled:             true,
+				IKSFileProviderName: "test-iks-file-provider",
+			},
+		},
+		{
+			name:           "VPC provider is enabled",
+			client:         &k8sClient,
+			expectedErrMsg: "",
+			// Setup test environment
+			vpcConfig: &config.VPCProviderConfig{
+				Enabled:       true,
+				VPCVolumeType: "test-vpc-volume-type",
+			},
+			iksConfig: nil,
+		},
+		{
+			name:   "No providers are enabled",
+			client: &k8sClient,
+			// Setup test environment
+			vpcConfig:      nil,
+			iksConfig:      nil,
+			expectedErrMsg: "no providers registered",
+		},
+		{
+			name: "pass nill k8s",
+			// Setup test environment
+			vpcConfig: &config.VPCProviderConfig{
+				Enabled:       true,
+				VPCVolumeType: "test-vpc-volume-type",
+			},
+			iksConfig:      nil,
+			client:         nil,
+			expectedErrMsg: "Description: Error initialising k8s client BackendError: validator: (nil *k8s_utils.KubernetesClient) ",
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			vpcfileconf.VPCConfig = tc.vpcConfig
+			vpcfileconf.IKSConfig = tc.iksConfig
+
+			_, err := InitProviders(vpcfileconf, tc.client, logger)
+			if tc.expectedErrMsg != "" {
+				assert.EqualError(t, err, tc.expectedErrMsg)
+			} else {
+				assert.NoError(t, err)
```

Comment:
same here

Reply:
We will take this in next PRs when we have more time.